src/main/scala/com/campudus/tableaux/arguments.scala

```diff
@@ -61,11 +73,13 @@ object ArgumentChecker {
 
   def checkArguments(args: ArgumentCheck[_]*): Unit = {
     val failedArgs: Vector[String] = args.zipWithIndex.foldLeft(Vector[String]()) {
-      case (v, (FailArg(ex), idx)) => v :+ s"($idx) ${ex.message}"
-      case (v, (OkArg(x), idx)) => v
+      case (errors, (FailArg(ex), idx)) => errors :+ s"($idx) ${ex.message}"
+      case (errors, (OkArg(x), idx)) => errors
     }
 
     if (failedArgs.nonEmpty) throw new IllegalArgumentException(failedArgs.mkString("\n"))
   }
 
+  def checked[A](arg1: ArgumentCheck[A]): A = arg1.get
+  def checked[A, B](arg1: ArgumentCheck[A], arg2: ArgumentCheck[B]): (A, B) = (arg1.get, arg2.get)
```

Comment:
We should definitely check out scalaz and use their approach for validation etc. I guess we wouldn't need the `ArgumentCheck` class at all...